src/System.Text.Json/src/System/Text/Json/JsonReader.cs

```diff
@@ -70,120 +89,218 @@ public bool Read()
             return _index < _str.Length;
         }
 
+        public Utf8String ReadWhiteSpace()
+        {
+            return ReadNextByte();
+        }
+
         public Utf8String ReadObjectStart()
         {
-            SkipAll();
-            var result = ReadToByte(CurlyOpenString, true);
-            GetNextTokenType();
-            return result;
+            return ReadNextByte();
         }
 
         public Utf8String ReadObjectEnd()
         {
-            SkipAll();
-            var result = ReadToByte(CurlyCloseString, true);
-            GetNextTokenType();
-            return result;
+            return ReadNextByte();
         }
 
         public Utf8String ReadArrayStart()
         {
-            SkipAll();
-            var result = ReadToByte(SquareOpenString, true);
-            GetNextTokenType();
-            return result;
+            return ReadNextByte();
         }
 
         public Utf8String ReadArrayEnd()
         {
-            SkipAll();
-            var result = ReadToByte(SquareCloseString, true);
-            GetNextTokenType();
-            return result;
+            return ReadNextByte();
         }
 
-        public Utf8String ReadProperty()
+        public Utf8String ReadComma()
         {
-            SkipAll();
-            // TODO: Use _str.SubstringFrom(new Utf8String("\"")).Substring(1).SubstringTo(new Utf8String("\""));
-            ReadToByte(QuoteString, true);
-            var result = ReadToByte(QuoteString, false);
-            GetNextTokenType();
-            return result;
+            return ReadNextByte();
         }
 
-        public long ReadPropertyValueAsInt()
+        public Utf8String ReadColon()
         {
-            SkipAll();
-            var isNegative = _str[_index] == DashString;
-            if (isNegative)
+            return ReadNextByte();
+        }
+
+        public bool ReadStart()
+        {
+            MoveToNextTokenType();
+            return true;
+        }
+
+        public bool ReadFinish()
+        {
+            MoveToNextTokenType();
+            return true;
+        }
+
+        public bool ReadTrue()
+        {
+            if (!Read())
+                throw new IndexOutOfRangeException("Json length is " + _str.Length + " and reading index " + _index +
+                                                   ".");
+            var utf8Bytes = new byte[4];
```

Comment:
Yes, you are right. The API I listed above is not yet available, but I can still use Substring(index, length) to achieve the same behavior without allocating.

Thanks.